src/charts/stacked-area.js

```diff
@@ -521,7 +523,10 @@ define(function(require){
          * @private
          */
         function cleanData(originalData) {
-            originalData = originalData.length === 0 ? createFakeData() : originalData;
+            originalData = 
+                originalData.length === 0 || shouldShowEmptyState
```

Comment:
Let´s get rid of "originalData.length" here, so only if shouldShowEmptyState is true we will show the empty state.